simpervisor/atexitasync.py

```diff
@@ -7,17 +7,39 @@
 import sys
 
 _handlers = []
+_existing_handlers = {}
 
 signal_handler_set = False
 
 
 def add_handler(handler):
     global signal_handler_set
     if not signal_handler_set:
```

Comment:
I'm not confident on `global` keyword use here.

`_handlers`, `_existing_handlers`, and `signal_handler_set` are all read and mutated in `add_handler`, but only `signal_handler_set` is declared in `global signal_handler_set` in `add_handler`.

I figure if the keyword is relevant for one, it should be relevant for all? Or, is it because `signal_handler_set` is possibly (not not actually in this code base) referenced from other .py files while `_handlers` and `_existing_handlers` aren't?

Reply:
Practical questions:
- do we need to add `global _handlers` and/or `global _existing_handlers` inside the `add_handler` function?
- could we remove `global signal_handler_set` in the `add_handler` function?

Reply:
Ah, got it!